### PR TITLE
Handle maintenance mode forms & deactivated forms

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes open-formulieren/open-forms#1967

* Form designers get a warning but can continue (maintenance mode)
* Everyone gets an error message for deactivated forms

### CODE REVIEW

File: src/components/RequireSubmission.js [hunk 1]
```diff
@@ -10,11 +13,23 @@ import { Redirect } from 'react-router-dom';
...
+
   if (!submission || !Object.keys(submission).length) {
     return (<Redirect to="/" />);
   }
   return (
-    <Component submission={submission} {...props} />
+    <>
+      {maintenanceModeAlert}
+      <Component submission={submission} {...props} />
+    </>
```
Comment (sergei-maertens) [inline src/components/RequireSubmission.js:32]: this is not ideal, I'd rather have it under the title, but that messes with the structure/re-usability of components so I took the path of least resistance.

File: src/components/RequireSubmission.js [hunk 2]
```diff
@@ -10,11 +13,23 @@ import { Redirect } from 'react-router-dom';
...
 const RequireSubmission = ({ submission, component: Component, ...props }) => {
+  const maintenanceMode = props?.form?.maintenanceMode;
+  const isFormDesigner = IsFormDesignerHeader.getValue();
+  let maintenanceModeAlert = null;
+  if (!isFormDesigner && maintenanceMode) {
+    throw new ServiceUnavailable('Service Unavailable', 503, 'Form in maintenance', 'form-maintenance');
+  } else if (isFormDesigner && maintenanceMode) {
+    maintenanceModeAlert = <MaintenanceMode />;
+  }
+
```
Comment (SilviaAmAm) [inline src/components/RequireSubmission.js:24]: Wouldn't it make more sense to do this after the check for whether there is a submission?

**Review by SilviaAmAm — CHANGES_REQUESTED** [review]

File: src/components/FormStart/index.js [hunk 3]
```diff
@@ -101,6 +108,8 @@ const FormStart = ({ form, onFormStart }) => {
     <LiteralsProvider literals={form.literals}>
       <Card title={form.name}>
 
+        {isFormDesigner && <MaintenanceMode asToast />}
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/components/FormStart/index.js:111]: I think you're right, I'll verify it!

File: src/components/FormStep.js [hunk 4]
```diff
@@ -66,12 +65,19 @@ const submitStepData = async (stepUrl, data) => {
...
+    try {
+      validateResponse = await post(validateUrl, data);
+    } catch(error) {
+      onBackendError(error);
+      next([{path: '', message: error.detail, code: error.code}]);
+      return;
+    }
 
     // process the errors
     if (validateResponse.status === 400) {
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/components/FormStep.js:83]: you're right again - I cleaned up some other unreachable code that similarly threw an error. I'll have to do type checking in the `catch` block and handle it appropriately.

Will try it again and fix it :-)

File: src/components/RequireSubmission.js (hunk 2 above)
Comment (sergei-maertens) [reply src/components/RequireSubmission.js:24]: It was originally like that, but then the page rendered the form step briefly (half a second or so), to then throw and go into the error boundary, which was a really janky user experience.

In terms of UX, if you can detect an error situation before the user even does any interaction (explicit or implicit), you want to handle it asap.

**Pull request merged**